File: UELPMAPP/src/app/administration/models/usermanagement.ts
import { UserProfile, GridDisplayInput, Companies, UserRoles, User } from "../../shared/models/shared.model";

export class UserManagement
{
    UserID:number;
    UserName:string;
    UserGUID:string;
    FirstName:string;
    LastName:string;
    CreatedDate:Date;
    LastUpdatedDate:Date;
    userAccountControl:string;
    Emailid:string;
    distinguishedName:string;
    logonCount?: number;
    primaryGroupID?: number;
    IsActive?:boolean;
    Thumbnail:ByteString;
    LocationId?:number;
    CompanyId?:number;
    ProfileImage:ByteString;
    Title:string;
    Password:string;
    isLocked?:boolean;
    Address1:string;
    Address2:string;
    Address3:string;
    CountryId?:number;    
    ZipCode:string;
    EmailSignature:string;
    PasswordAttemptCount?:number;
    AlterApprovarUserId?:number;
    ApprovalStartDate?:Date;
    ApprovalEndDate?:Date;
    Designation:string;
    PhoneNumber:string;
    RolesDetails:Array<RolesDetails>;
    RolesDetailsToDelete:Array<number>;
    CountryName:string;
    User:User;
    ManagerId:number;
    ManagerName:string;
    CreatedBy:number;
    isADUser:boolean;
    NewPassword:string;
    IsWorkFlowAssigned: boolean;
}

// export class UserManagementList
// {
//     UserId:number;
//     UserName :string;
//     FirstName: string;
//     LastName: string;
//     UserRoles:number ;
//     UserRolesName :string;
// }

export class UserManagementList
{
    UserID:number;
    FirstName :string;
    EmailId :string;
    PhoneNumber :string;
    Designation :string;
}

export class UserManagementDisplayResult
{
    UserManagementList:Array<UserManagementList>;
    TotalRecords :number;
}

export class UserManagementFilterDisplayInput extends GridDisplayInput
{
    UserNameFilter:string;
    RolesNameFilter :string;
}

export class RolesDetails
{
    UserRoleID:number;
    UserID:number;
    Company:Companies;
    Role:UserRoles;
    constructor()
    {
        this.UserRoleID=0;
        this.UserID=0;
        this.Company={
            CompanyId:0,
            CompanyName:"",
            SupplierVerifier:0,
            IsSelected:0,
            RoleID:0
        }
        this.Role={
            RoleID:0,
            RoleName:"",
            IsSelected: false
        }
    }
}

File: UELPMAPI/UELPM.Model/Models/UserManagement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UELPM.Model.Models
{
    public class UserManagement
    {
        public int UserID { get; set; }
        public string UserName { get; set; }
        public string UserGUID { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? LastUpdatedDate { get; set; }
        public string userAccountControl { get; set; }
        public string Emailid { get; set; }
        public string distinguishedName { get; set; }
        public int? logonCount { get; set; }
        public int? primaryGroupID { get; set; }
        public bool? IsActive { get; set; }
        public byte[] Thumbnail { get; set; }
        public int? LocationId { get; set; }
        public int? CompanyId { get; set; }
        public byte[] ProfileImage { get; set; }
        public string Title { get; set; }

        public string CurrentPassword { get; set; }
        public string Password { get; set; }
        public bool? isLocked { get; set; }
        public string Address1 { get; set; }
        public string Address2 { get; set; }
        public string Address3 { get; set; }
        public int CountryId { get; set; }
        public string ZipCode { get; set; }
        public string EmailSignature { get; set; }
        public int? PasswordAttemptCount { get; set; }
        public int? AlterApprovarUserId { get; set; }
        public DateTime? ApprovalStartDate { get; set; }
        public DateTime? ApprovalEndDate { get; set; }
        public string Designation { get; set; }
        public string PhoneNumber { get; set; }       
        public List<RolesDetails> RolesDetails { get; set; }
        public List<int> RolesDetailsToDelete { get; set; }
        public Country Country { get; set; }
        public string CountryName { get; set; }
        public User User { get; set; }
        public int CreatedBy { get; set; }
        public bool isADUser { get; set; }
        public string NewPassword { get; set; }
        public bool IsWorkFlowAssigned { get; set; }

        public int ManagerId { get; set; }

        public string ManagerName { get; set; }
    }


    public class RolesDetails
    {
        public int UserRoleId { get; set; }
        public int UserID { get; set; }
        public Companies Company { get; set; }
        public List<Locations> DepartmentList { get; set; }
        public List<UserRoles> Role { get; set; }
        public bool IsSelected { get; set; }
    }

    public class UserDepartment : Locations
    {
        public int UserId { get; set; }
        public int CompanyId { get; set; }
    }

    public class UserCompany
    {
        public int UserId { get; set; }
        public int CompanyId { get; set; }
        public bool IsSelected { get; set; }
        public string CompanyName { get; set; }
    }

    public class UserRole : UserRoles
    {
        public int UserId { get; set; }
        public int CompanyId { get; set; }
    }

    public class UserManagementDisplayResult
    {
        public List<UserManagementList> UserManagementList { get; set; }
        public int TotalRecords { get; set; }
    }

    public class UserManagementList
    {
        public int UserID { get; set; }

        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string EmailId { get; set; }
        public string PhoneNumber { get; set; }
        public string Designation { get; set; }

        
    }

    public class ValidateUserManagement
    {
        public int UserId { get; set; }
    }

    public class UserManagementFilterDisplayInput :GridDisplayInput
    {
        public string UserNameFilter { get; set; }
        public string RolesNameFilter { get; set; }
    }

}
